### PR TITLE
fix(evolution): support date column inside groups

### PR DESCRIPTION
**Pull request by PrettyWood**

### CODE REVIEW

File: server/tests/backends/mongo_translator/test_mongo_translator_steps.py [hunk 1]
```diff
@@ -15,6 +15,14 @@
...
 exec_type = "mongo"
 test_cases = retrieve_case("mongo_translator", exec_type)
+exec_step = ""  # all steps
+# exec_step = "evolution"
+exec_fixture_names = [""]  # all fixtures
+# exec_fixture_names = [
+#     "abs_with_same_group",
+#     "multiple_evolutions_with_groups",
+# ]
+test_cases = [tc for tc in test_cases if exec_step in tc.id and any(f_name in tc.id for f_name in exec_fixture_names)]
```
Comment (PrettyWood) [inline server/tests/backends/mongo_translator/test_mongo_translator_steps.py:25]: j'ai laissé car c'est qd meme pratique (en tout cas pour moi ca l'a été)
Après on voudrait surement ajouter un `pytest_addoption` plutôt

File: server/tests/backends/fixtures/evolution/abs_with_same_group.json
Comment (lukapeschke) [inline server/tests/backends/fixtures/evolution/abs_with_same_group.json:1]: Fixture LGTM, but could you please use YAML and add a comment explaining that we're using `DATE` in both `date_col` and `index_columns` ?

File: server/pyproject.toml [hunk 2]
```diff
@@ -71,9 +71,7 @@ allow_redefinition = true
 check_untyped_defs = true
 
 [tool.pytest.ini_options]
-env = [
-    "PYTHONPATH = ."
-]
+pythonpath = "."
```
Comment (lukapeschke) [inline server/pyproject.toml:74]: might be annoying to cherry-pick, no ?

File: server/tests/backends/mongo_translator/test_mongo_translator_steps.py
Comment (lukapeschke) [inline server/tests/backends/mongo_translator/test_mongo_translator_steps.py:1]: to undo

File: server/tests/backends/fixtures/evolution/abs_with_same_group.json
Comment (lukapeschke) [inline server/tests/backends/fixtures/evolution/abs_with_same_group.json:1]: Fixture LGTM but could you please format it as YAMl and add a comment explaining the specificity of the fixture ?

**Review by lukapeschke — COMMENTED** [review]
A few nits but LGTM, could you please also add an entry to `server/CHANGELOG.md` ?

**Review by lukapeschke — APPROVED** [review]

**Pull request merged**